### PR TITLE
fixes child pipeline jobs not found

### LINKED ISSUE #344: 404 Not found for child pipelinejobs
We have a pipeline for tags which triggers child pipelines. I noticed 404 errors for this use case in logfile:
`{"error":"GET https://gitlab.com/api/v4/projects/group/project/pipelines/3835447/jobs: 404 {message: 404 Not found}","level":"warning","msg":"pulling ref metrics","project-name":"group/project","ref":"v1.1","time":"2021-10-08T09:44:00Z"}`

This does only happen when `ref.Project.Pull.Pipeline.Jobs.FromChildPipelines.Enabled`is true.
I expect those child jobs will be missing, but i did not check it and it might probably happen for other refs as well.

The pipeline id in this URL is actually the child pipeline on the child-project, so getting a 404 is correct in this case. 

I was able to debug it to: https://github.com/mvisonneau/gitlab-ci-pipelines-exporter/blob/013e6431bb664907bff7ce5080e122c9420204e9/pkg/gitlab/jobs.go#L145 
and later to https://github.com/mvisonneau/gitlab-ci-pipelines-exporter/blob/013e6431bb664907bff7ce5080e122c9420204e9/pkg/gitlab/jobs.go#L130 
The parent project name seems wrong here

### PR DESCRIPTION
**Pull request by vgropp**
resolves #344

### CODE REVIEW

File: pkg/gitlab/jobs.go [hunk 1]
```diff
@@ -116,18 +116,23 @@ func (c *Client) ListPipelineBridges(projectName string, pipelineID int) (bridge
 
 // ListPipelineChildJobs ..
 func (c *Client) ListPipelineChildJobs(projectName string, parentPipelineID int) (jobs []schemas.Job, err error) {
-	pipelineIDs := []int{parentPipelineID}
+	type piplineDef struct {
```
Comment (mvisonneau) [inline pkg/gitlab/jobs.go:119]:
```suggestion
	type pipelineDef struct {
```

File: pkg/gitlab/jobs.go [hunk 2]
```diff
@@ -116,18 +116,23 @@ func (c *Client) ListPipelineBridges(projectName string, pipelineID int) (bridge
...
+		projectName interface{}
```
Comment (mvisonneau) [inline pkg/gitlab/jobs.go:120]:
```suggestion
		projectNameOrID interface{}
```

File: pkg/gitlab/jobs.go [hunk 3]
```diff
@@ -116,18 +116,23 @@ func (c *Client) ListPipelineBridges(projectName string, pipelineID int) (bridge
...
+		pipelineID  int
+	}
+
+	pipelines := []piplineDef{{projectName, parentPipelineID}}
```
Comment (mvisonneau) [inline pkg/gitlab/jobs.go:124]:
```suggestion
	pipelines := []pipelineDef{{projectNameOrID, parentPipelineID}}
```

File: pkg/gitlab/jobs.go [hunk 4]
```diff
@@ -78,7 +78,7 @@ func (c *Client) ListPipelineJobs(projectName string, pipelineID int) (jobs []sc
 }
 
 // ListPipelineBridges ..
-func (c *Client) ListPipelineBridges(projectName string, pipelineID int) (bridges []*goGitlab.Bridge, err error) {
+func (c *Client) ListPipelineBridges(projectName interface{}, pipelineID int) (bridges []*goGitlab.Bridge, err error) {
```
Comment (mvisonneau) [inline pkg/gitlab/jobs.go:81]:
```suggestion
func (c *Client) ListPipelineBridges(projectNameOrID interface{}, pipelineID int) (bridges []*goGitlab.Bridge, err error) {
```

File: pkg/gitlab/jobs.go [hunk 5]
```diff
@@ -140,9 +145,9 @@ func (c *Client) ListPipelineChildJobs(projectName string, parentPipelineID int)
 				continue
 			}
 
-			pipelineIDs = append(pipelineIDs, foundBridge.DownstreamPipeline.ID)
+			pipelines = append(pipelines, piplineDef{foundBridge.DownstreamPipeline.ProjectID, foundBridge.DownstreamPipeline.ID})
```
Comment (mvisonneau) [inline pkg/gitlab/jobs.go:148]:
```suggestion
			pipelines = append(pipelines, pipelineDef{foundBridge.DownstreamPipeline.ProjectID, foundBridge.DownstreamPipeline.ID})
```

File: pkg/gitlab/jobs.go [hunk 6]
```diff
@@ -39,7 +39,7 @@ func (c *Client) ListRefPipelineJobs(ref schemas.Ref) (jobs []schemas.Job, err e
 }
 
 // ListPipelineJobs ..
-func (c *Client) ListPipelineJobs(projectName string, pipelineID int) (jobs []schemas.Job, err error) {
+func (c *Client) ListPipelineJobs(projectName interface{}, pipelineID int) (jobs []schemas.Job, err error) {
```
Comment (mvisonneau) [inline pkg/gitlab/jobs.go:42]:
```suggestion
func (c *Client) ListPipelineJobs(projectNameOrID interface{}, pipelineID int) (jobs []schemas.Job, err error) {
```

**Comment by mvisonneau** [comment]
Thanks for the changes @vgropp, I am looking to spend a bit more time on it before merging it as I would like to find a way to avoid having multiple entities being held for a single project (referenced with either its name or ID)

**Comment by vgropp** [comment]
yes, take your time or maybe with another idea. i must confess i did not take a deep look how it would show up as I expected it only a bug with this feature

**Comment by vgropp** [comment]
@mvisonneau any news on this, we would like to use this

**Comment by mvisonneau** [comment]
@vgropp I rebased and merged throughf 003ece95f57e9e9b49136850653b3c83308b8a5, thanks again for this fix! 🙇

**Comment by vgropp** [comment]
great, thanks for the fast reaction 👍

**Pull request closed without merging**